service/src/db/dbOperations.ts

```diff
@@ -22,9 +22,30 @@ export async function findResourcesWithQuery<T extends fhir4.FhirResource>(
   const collection = Connection.db.collection(resourceType);
   query._dataRequirements = { $exists: false };
   query._summary = { $exists: false };
+  query._elements = { $exists: false };
   return collection.find<T>(query, { projection: { _id: 0, _dataRequirements: 0 } }).toArray();
 }
 
+/**
+ * searches the database and returns an array of all resources of the given type that match the given query
+ * but the resources only include the elements specified by the _elements parameter
+ */
+export async function findResourceElementsWithQuery<T extends fhir4.FhirResource>(
+  query: Filter<any>,
+  resourceType: FhirResourceType
+) {
+  const collection = Connection.db.collection(resourceType);
+  const projection: any = { status: 1, resourceType: 1, type: 1 };
```

Comment:
Ugh this also drives me nuts. So `type` is actually a required field in a `Library` resource but not a `Measure` resource. Maybe I was just being lazy, but I figured there wasn't really any harm in just included it in both rather than doing type checking, what do you think?

Reply:
Looking at the `_elements` spec:

> Only elements that are listed are to be returned. Clients SHOULD list all mandatory and modifier elements in a resource as part of the list of elements. Servers have the right to return more data than request, however clients can generally expect that returned elements will be:
> 
> - elements explicity requested,
> - mandatory elements, or
> - modifier elements that have values.

Kind of feels like we should do type checking :/

Reply:
Also looking at the spec, looks like we might need the [SUBSETTED](https://terminology.hl7.org/CodeSystem-v3-ObservationValue.html#v3-ObservationValue-SUBSETTED) flag?

Reply:
Always including the required elements makes sense.

I agree with the tagging thing. We should add `SUBSETTED` to the [meta.tag](https://hl7.org/fhir/R4/resource-definitions.html#Meta.tag) on the resources we send back with `_elements`.